EstateReporting.IntegrationTests/projections/continuous/MerchantBalanceCalculator.js

```diff
@@ -0,0 +1,148 @@
+var fromStreams = fromStreams || require('../../node_modules/event-store-projection-testing').scope.fromStreams;
+var partitionBy = partitionBy !== null ? partitionBy : require('../../node_modules/event-store-projection-testing').scope.partitionBy;
+var emit = emit || require('../../node_modules/event-store-projection-testing').scope.emit;
+
+var incrementBalanceFromDeposit = function(s, merchantId, amount, dateTime) {
+    var merchant = s.merchants[merchantId];
+    merchant.Balance += amount;
+    merchant.AvailableBalance += amount;
+        // protect against events coming in out of order
+    if (merchant.LastDepositDateTime === null || dateTime > merchant.LastDepositDateTime) {
+            merchant.LastDepositDateTime = dateTime;
+        }
+    }
+
+var addPendingBalanceUpdate = function(s, merchantId, amount, transactionId, dateTime)
+{
+    var merchant = s.merchants[merchantId];
+    merchant.AvailableBalance -= amount;
+    merchant.PendingBalanceUpdates[transactionId] = {
+        Amount: amount,
+        TransactionId: transactionId
+    };
+    // protect against events coming in out of order
+    if (merchant.LastSaleDateTime === null || dateTime > merchant.LastSaleDateTime)
+    {
+        merchant.LastSaleDateTime = dateTime;
+    }
+}
+
+var decrementBalanceForSale = function(s, merchantId, transactionId, isAuthorised)
+{
+    var merchant = s.merchants[merchantId];
+    // lookup the balance update
+    var balanceUpdate = merchant.PendingBalanceUpdates[transactionId];
+
+    if (balanceUpdate !== undefined)
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/manual/stuart_ferguson1/EstateReporting/pullRequest?prid=6339123)